cloudformation/infrastructure/fargate-cluster.yaml

```diff
@@ -0,0 +1,282 @@
+              - 'kms:GenerateDataKey'
+              - 's3:PutObject'
+              - 's3:GetObject'
+              - 's3:AbortMultipartUpload'
+              - 's3:ListMultipartUploadParts'
+              - 's3:ListBucket'
+              - 's3:ListBucketMultipartUploads'              
+              Resource:
+                - "*"
+
+  ConcordiaEC2Role:
+    Type: AWS::IAM::Role
+    Properties:
+      Path: /
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: ec2.amazonaws.com
+            Action: sts:AssumeRole
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly
+        - arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy
+  
+  ConcordiaInstanceProfile:
+    Type: AWS::IAM::InstanceProfile
+    Properties:
+      Path: /
+      Roles:
+        - !Ref 'ConcordiaEC2Role'
+  
+  ConcordiaTaskRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: ecs-tasks.amazonaws.com
+            Action:
+              - sts:AssumeRole              
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly
+        - arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy
+  
+  ConcordiaAppLogsGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Ref AWS::StackName
+      RetentionInDays: 30
+
+  ConcordiaExternalTargetGroup:
+    Type: AWS::ElasticLoadBalancingV2::TargetGroup
+    Properties:
+      HealthCheckIntervalSeconds: 10
+      HealthCheckPath: /healthz/
+      HealthCheckProtocol: HTTP
+      HealthCheckTimeoutSeconds: 5
+      HealthyThresholdCount: 2
+      UnhealthyThresholdCount: 8
+      TargetType: ip
+      Port: 80
+      Protocol: HTTP
+      VpcId: !Ref VpcId
+
+  LoadBalancer:
+    Type: AWS::ElasticLoadBalancingV2::LoadBalancer
+    Properties:
+        Name: !Ref EnvironmentName
+        Subnets: !Ref PublicSubnets
+        SecurityGroups: 
+            - !Ref LoadBalancerSecurityGroup
+        Tags: 
+            - Key: Name
+              Value: !Ref EnvironmentName
+
+  ExternalLoadBalancerListener:
+    Properties:
+      DefaultActions:
+        - TargetGroupArn: !Ref 'ConcordiaExternalTargetGroup'
+          Type: forward
+      LoadBalancerArn:  !Ref LoadBalancer
+      Port: 80
+      Protocol: HTTP
+    Type: AWS::ElasticLoadBalancingV2::Listener
+
+  HttpListenerRule:
+    Type: AWS::ElasticLoadBalancingV2::ListenerRule
+    Properties:
+      ListenerArn: !Ref ExternalLoadBalancerListener
+      Priority: 1
+      Conditions:
+          - Field: path-pattern
+            Values:
+                - /*
+      Actions:
+          - TargetGroupArn: !Ref ConcordiaExternalTargetGroup
+            Type: forward
+
```

Comment:
The SSL listener rule will have to be added manually because getting the cert is done manually

Reply:
Can't we reference a pre-defined cert in IAM? I'm assuming that'll be something like `!Sub 'arn:aws:iam::${AWS::AccountId}':server-certificate/${CanonicalHostName}`, assuming that we also set a property which defines the public hostname we want to use which would also be handy to pass to the container to seed the sites framework.